.github/workflows/ui-tests.yml

```
@@ -143,158 +63,30 @@ jobs:
-        uses: mikepenz/action-junit-report@65fe03598d8d251738592a497a9e8547a5c48eaa # v5.6.0
-        if: always()
-        with:
-          report_paths: "build/reports/junit.xml"
-          fail_on_failure: true
-          fail_on_parse_error: true
-          detailed_summary: true
-
-      - name: Upload Result Bundle
-        uses: actions/upload-artifact@v4
-        if: ${{ failure() }}
-        with:
-          name: ui-tests-ios-swift6.xcresult
-          path: fastlane/test_results/ui-tests-ios-swift6.xcresult
-
-      - name: Archiving Raw Test Logs
-        uses: actions/upload-artifact@v4
-        if: ${{ failure() || cancelled() }}
-        with:
-          name: ui-tests-ios-swift6-raw-output
-          path: |
-            ~/Library/Logs/scan/*.log
-            ./fastlane/test_output/**
-
-      - name: Store screenshot
-        uses: ./.github/actions/capture-screenshot
-        if: failure()
+    uses: ./.github/workflows/ui-tests-common.yml
+    with:
+      fastlane_command: ui_tests_ios_swift6
+      xcresult_name: ui-tests-ios-swift6
+      xcode_version: 16.2
+      build_with_make: true
+      macos_version: macos-15
 
   duplication-tests:
     name: UI Tests for project with Sentry duplicated - V4 # Up the version with every change to keep track of flaky tests
-    runs-on: macos-15
-
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Print hardware info
-        run: system_profiler SPHardwareDataType
-      - uses: ruby/setup-ruby@v1
-        with:
-          bundler-cache: true
-
-      - run: ./scripts/ci-select-xcode.sh "16.4"
-
-      - run: ./scripts/build-xcframework.sh gameOnly
-
-      - name: Add Microphone permissions
-        uses: ./.github/actions/add-microphone-permissions
-
-      - name: Run Fastlane
-        run: bundle exec fastlane duplication_test
-
-      - name: Publish Test Report
-        uses: mikepenz/action-junit-report@65fe03598d8d251738592a497a9e8547a5c48eaa # v5.6.0
-        if: always()
-        with:
-          report_paths: "build/reports/junit.xml"
-          fail_on_failure: true
-          fail_on_parse_error: true
-          detailed_summary: true
-
-      - name: Upload Result Bundle
-        uses: actions/upload-artifact@v4
-        if: ${{ failure() }}
-        with:
-          name: duplication-test.xcresult
-          path: fastlane/test_results/duplication-test.xcresult
-
-      - name: Archiving Raw Test Logs
-        uses: actions/upload-artifact@v4
-        if: ${{ failure() || cancelled() }}
-        with:
-          name: duplication-test-raw-output
-          path: |
-            ~/Library/Logs/scan/*.log
-            ./fastlane/test_output/**
-
-      - name: Store screenshot
-        uses: ./.github/actions/capture-screenshot
-        if: failure()
+    uses: ./.github/workflows/ui-tests-common.yml
+    with:
+      fastlane_command: duplication_test
+      xcresult_name: duplication-test
+      xcode_version: 16.4
+      build_with_script: true
```

here's the other one
```suggestion
      build_xcframework: true
```